frontend/src/App.jsx

```diff
@@ -1,8 +1,33 @@
-import React from "react";
-import Loader from "./Components/Loader/Loader";
+import React, { useState } from "react";
+import { Routes, Route } from "react-router-dom";
+import HomePage from "./Components/HomePage/HomePage";
+import Login from "./Components/Login/Login";
+import SignUp from "./Components/SignUp/SignUp";
+import ProtectedRoute from "./Components/Routes/ProtectedRoute";
+import RedirectRoute from "./Components/Routes/RedirectRoute";
 
 const App = () => {
-	return <Loader />;
+	const [accessToken, setAccessToken] = useState(localStorage.accessToken);
+
+	return (
+		<Routes>
+			<Route path="/" exact element={<RedirectRoute accessToken={accessToken} />} />
+			<Route path="login" exact element={<Login setAccessToken={setAccessToken} />} />
+			<Route path="signup" exact element={<SignUp />} />
+			<Route
+				path="homepage"
+				exact
+				element={
+					<ProtectedRoute accessToken={accessToken}>
+						<HomePage setAccessToken={setAccessToken} />
+					</ProtectedRoute>
+				}
+			/>
+
+			{/* Other Paths */}
+			<Route path="*" element={<RedirectRoute accessToken={accessToken} />} />
+		</Routes>
```

Comment:
Looks good 👍 . Next step is to avoid passing setAccessToken and accessToken and place it in context/redux/whatever you choose.

Reply:
That is something we will implement as soon as we're satisfied with base functionality and everything else we wanted to implement, ofc if you think that this is crucial we'll implement it right away.